src/queries/SUE/configurations.tsx

```diff
@@ -0,0 +1,43 @@
+import _camelCase from 'lodash/camelCase';
+import _mapKeys from 'lodash/mapKeys';
+
+import { fetchSueEndpoints } from '../../helpers';
+
+export const configurations = async () => {
+  const {
+    sueConfigurationsUrl = '',
+    sueContactRequiredFieldConfigurationUrl = '',
+    sueFetchObj = {},
+    sueGeoMapConfigurationUrl = ''
+  } = await fetchSueEndpoints();
+
+  const configPromise = await (await fetch(`${sueConfigurationsUrl}`, sueFetchObj)).json();
+  const geoMapPromise = await (await fetch(`${sueGeoMapConfigurationUrl}`, sueFetchObj)).json();
+  const requiredFieldPromise = await (
+    await fetch(`${sueContactRequiredFieldConfigurationUrl}`, sueFetchObj)
+  ).json();
+
+  return Promise.all([configPromise, geoMapPromise, requiredFieldPromise])
+    .then(([configResponse, geoMapResponse, requiredFieldResponse]) => {
```

Comment:
what about some refactoring here. i suggest something more readable without that much awaits chained. we can have some more code to optimize understandment.

---

a ChatGPT summary:

Here's a refactored version of your code that optimizes readability and maintains efficiency by handling the fetch requests in parallel without chaining unnecessary `await` statements on the `.json()` calls:

```javascript
async function fetchConfigurationData() {
  // Use Promise.all to handle parallel fetch requests
  const urls = [
    sueConfigurationsUrl,
    sueGeoMapConfigurationUrl,
    sueContactRequiredFieldConfigurationUrl
  ];

  // Map over the URLs and fetch each configuration
  const fetchPromises = urls.map(url => fetch(url, sueFetchObj));

  try {
    // Wait for all fetch requests to resolve
    const responses = await Promise.all(fetchPromises);

    // Convert each response to JSON
    const jsonDataPromises = responses.map(response => response.json());

    // Resolve all JSON conversion promises
    const [configResponse, geoMapResponse, requiredFieldResponse] = await Promise.all(jsonDataPromises);

    // Handle the resolved data as needed
    return { configResponse, geoMapResponse, requiredFieldResponse };
  } catch (error) {
    // Handle errors, such as network issues or invalid JSON
    console.error('Failed to fetch configurations:', error);
    throw error;  // Rethrow or handle error as needed
  }
}
```

#### Changes and Improvements:
1. **Parallel Fetches:** This refactor ensures that the initial `fetch` requests are sent out in parallel, without awaiting each one sequentially.
2. **Promise Handling:** The `.json()` method is called on each response after all fetch requests have resolved, which maintains parallelism in processing responses as well.
3. **Error Handling:** The try-catch block catches errors from either the fetch or JSON parsing phase and handles them appropriately, making the function more robust.
4. **Clean Structure:** Using functions and separating concerns makes the code more modular and easier to read or modify.

---

please have a closer look and try different solutions as well to find the one that fits well for us.

Reply:
This code doesn't do exactly what we want. But it is much more understandable. So I merged the two codes together and came up with a code that is more understandable and does the same thing.

https://github.com/smart-village-solutions/smart-village-app-app/pull/1131/commits/8bed36e034809bec34bcfb61029f917f0feabfa4